src/idl_gen_js.cpp

```diff
@@ -59,7 +65,14 @@ class JsGenerator : public BaseGenerator {
       code += exports_code;
     }
 
-    return SaveFile(GeneratedFileName(path_, file_name_).c_str(), code, false);
+    std::string filename = "";
+    if (ts) {
```

Comment:
yes, all these if-thens can use some cleanup. prefer to stick them in a table (see `LanguageParameters` in `idl_gen_general.cpp` for an example) where possible, and otherwise think how to represent the code with as few if-thens as possible by sticking them into functions where needed.